File: Medium/1877. Minimize Maximum Pair Sum in Array/1877. Minimize Maximum Pair Sum in Array.java
import java.util.Arrays;

class Solution {
    public int minPairSum(int[] nums) {
        Arrays.sort(nums);

        int maxPairSum = 0;
        int n = nums.length;

        for (int i = 0; i < n / 2; i++) {
            maxPairSum = Math.max(maxPairSum, nums[i] + nums[n - i - 1]);
        }

        return maxPairSum;
    }
}

File: Medium/1877. Minimize Maximum Pair Sum in Array/1877. Minimize Maximum Pair Sum in Array.rs
impl Solution {
    pub fn min_pair_sum(nums: Vec<i32>) -> i32 {
        let mut nums = nums;
        nums.sort();

        let mut max_pair_sum = 0;
        let n = nums.len();

        for i in 0..n / 2 {
            max_pair_sum = max_pair_sum.max(nums[i] + nums[n - i - 1]);
        }

        max_pair_sum
    }
}
